source/simulator/melt.cc

```diff
@@ -1071,53 +1073,6 @@ namespace aspect
       // u_f =  u_s - K_D (nabla p_f - rho_f g) / phi  or = 0
       // by solving a mass matrix problem
 
-      // TODO: lots of cleanup/optimization opportunities here:
```

Comment:
I am missing the code to zero out the matrix and RHS or is this hidden somewhere?

Reply:
No I think it was missing. I added it now. I think it did not make a difference, because whenever we rebuild the Stokes matrix we zero out the whole system matrix in `assembly.cc:assemble_Stokes_system()`. Not sure that is how it should be (the advection system only zeroes out its own block).

Reply:
I think each assemble routine should zero out exactly the blocks we are writing to.

Reply:
I can open another PR for the Stokes assembly.